fn_ansible/fn_ansible/lib/ansible_api.py

```diff
@@ -0,0 +1,106 @@
+    and returns the results of the queries that the playbook 
+    contains.
+    """
+       
+    loader = DataLoader()
+    if(vault_password_file):
+        vault_password = read_secret(vault_password_file, VAULT_PASS_KEY)
+        loader.set_vault_secrets([(DEFAULT_VAULT_ID, VaultSecret(_bytes=to_bytes(vault_password)))])
+    inventory = InventoryManager(loader=loader, sources=hosts_path)
+    variable_manager = VariableManager(loader=loader, inventory=inventory)
+    if playbook_extra_vars:
+        variable_manager.extra_vars = playbook_extra_vars
+
+    passwords={'become_pass': root_password}
+
+    Options = namedtuple('Options',
+                        ['connection',
+                        'remote_user',
+                        'ask_sudo_pass',
+                        'verbosity',
+                        'ack_pass',
+                        'module_path',
+                        'forks',
+                        'become',
+                        'become_method',
+                        'become_user',
+                        'check',
+                        'listhosts',
+                        'listtasks',
+                        'listtags',
+                        'syntax',
+                        'sudo_user',
+                        'sudo',
+                        'diff'])
+    options = Options(connection=connection_type,
+                        remote_user=user_name,
+                        ack_pass=None,
+                        sudo_user=None,
+                        forks=NUM_OF_FORKS,
+                        sudo=None,
+                        ask_sudo_pass=False,
+                        verbosity=5,
+                        module_path=None,
+                        become=True,
+                        become_method=playbook_become_method,
+                        become_user=playbook_become_user,
+                        check=False,
+                        diff=False,
+                        listhosts=None,
+                        listtasks=None,
+                        listtags=None,
+                        syntax=None)
+
+    playbook = PlaybookExecutor(playbooks=playbook_path,inventory=inventory,
+                variable_manager=variable_manager,
+                loader=loader,options=options,passwords=passwords)
+    
+    result = {}
+    try:
+        playbook.run()
+        stats = playbook._tqm._stats
+        hosts = sorted(stats.processed.keys())
+        for h in hosts:
+            result = {h: stats.summarize(h)}
+    except Exception as e:
+        original_error = traceback.format_exc(e)
+        caution_text = "> That didn't work, please check 3 important things: \
+        \n\t > 1. Configuration in app.config file \n\t \
+        > 2. Playbook variables provided in correct format without extra space \
+        \n\t > 3. Playbook syntax."
+        raise ValueError(original_error + "\n\t" + caution_text)
+    
+    return result
+
+
+def read_secret(file,key):
```

Comment:
* Avoid using the variable name `file` as it is a predefined variable in Python: https://docs.python.org/2.7/library/functions.html?highlight=file#file